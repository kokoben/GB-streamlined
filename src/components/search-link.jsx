import React from 'react';
import { Icon } from 'antd';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

const SearchLink = props => (
  props.navDirection === 'previous' ? (
    <div
      role="button"
      tabIndex={0}
      onClick={(e) => {
        e.stopPropagation();
        props.navLink(props.currentSearchPage - 1);
      }}
      onKeyDown={(e) => {
        if (e.key === 'Enter') {
          e.stopPropagation();
          props.navLink(props.currentSearchPage - 1);
        }
      }}
    >
      <Icon type="up-square-o" />
      <span style={{ paddingLeft: '5px' }}>
        See previous results
      </span>
    </div>
  ) : (
    <div
      role="button"
      tabIndex={0}
      onClick={(e) => {
        e.stopPropagation();
        props.navLink(props.currentSearchPage + 1);
      }}
      onKeyDown={(e) => {
        if (e.key === 'Enter') {
          e.stopPropagation();
          props.navLink(props.currentSearchPage + 1);
        }
      }}
    >
      <span style={{ paddingRight: '5px' }}>
        See more results
      </span>
      <Icon type="down-square-o" />
    </div>
  )
);

SearchLink.propTypes = {
  currentSearchPage: PropTypes.number.isRequired,
  navDirection: PropTypes.string.isRequired,
  navLink: PropTypes.func.isRequired,
};

const mapStateToProps = state => ({
  currentSearchPage: state.shared.currentSearchPage,
});

export default connect(mapStateToProps, null)(SearchLink);
