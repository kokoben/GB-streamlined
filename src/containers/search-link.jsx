import React from 'react';
import { Icon } from 'antd';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import PropTypes from 'prop-types';
import { setSearchPage, setSearchMarker } from '../actions/index';

export const SearchLink = props => (
  props.navDirection === 'previous' ? (
    <div
      role="button"
      tabIndex={0}
      onClick={(e) => {
        e.stopPropagation();
        props.setSearchPage(props.currentSearchPage - 1);
        props.setSearchMarker(props.searchResultMarker - 8);
      }}
      onKeyDown={(e) => {
        if (e.key === 'Enter') {
          e.stopPropagation();
          props.setSearchPage(props.currentSearchPage - 1);
          props.setSearchMarker(props.searchResultMarker - 8);
        }
      }}
    >
      <Icon type="up-square-o" />
      <span style={{ paddingLeft: '5px' }}>
        See previous results
      </span>
    </div>
  ) : (
    <div
      role="button"
      tabIndex={0}
      onClick={(e) => {
        e.stopPropagation();
        props.setSearchPage(props.currentSearchPage + 1);
        props.setSearchMarker(props.searchResultMarker + 8);
      }}
      onKeyDown={(e) => {
        if (e.key === 'Enter') {
          e.stopPropagation();
          props.setSearchPage(props.currentSearchPage + 1);
          props.setSearchMarker(props.searchResultMarker + 8);
        }
      }}
    >
      <span style={{ paddingRight: '5px' }}>
        See more results
      </span>
      <Icon type="down-square-o" />
    </div>
  )
);

SearchLink.propTypes = {
  currentSearchPage: PropTypes.number.isRequired,
  searchResultMarker: PropTypes.number.isRequired,
  navDirection: PropTypes.string.isRequired,
  setSearchPage: PropTypes.func.isRequired,
  setSearchMarker: PropTypes.func.isRequired,
};

const mapStateToProps = state => ({
  currentSearchPage: state.shared.currentSearchPage,
  searchResultMarker: state.shared.searchResultMarker,
});

const mapDispatchToProps = dispatch => (
  bindActionCreators({
    setSearchPage,
    setSearchMarker,
  }, dispatch)
);

export default connect(mapStateToProps, mapDispatchToProps)(SearchLink);
