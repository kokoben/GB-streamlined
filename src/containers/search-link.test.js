import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { SearchLink } from './search-link';

it('renders correctly', () => {
  const mockSetSearchPage = jest.fn();
  const mockSetSearchMarker = jest.fn();

  const wrapper = shallow(<SearchLink
      setSearchPage={mockSetSearchPage}
      setSearchMarker={mockSetSearchMarker}
      currentSearchPage={1}
      searchResultMarker={1}
      navDirection="prev"
  />);
  const tree = toJson(wrapper);
  expect(tree).toMatchSnapshot();
});
