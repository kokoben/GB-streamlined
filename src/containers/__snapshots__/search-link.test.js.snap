// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  onClick={[Function]}
  onKeyDown={[Function]}
  role="button"
  tabIndex={0}
>
  <span
    style={
      Object {
        "paddingRight": "5px",
      }
    }
  >
    See more results
  </span>
  <Icon
    type="down-square-o"
  />
</div>
`;
